src/main/java/com/github/stefvanschie/inventoryframework/GuiListener.java

```diff
@@ -3,23 +3,53 @@
 import com.github.stefvanschie.inventoryframework.pane.Pane;
 import org.bukkit.entity.HumanEntity;
 import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.inventory.InventoryClickEvent;
 import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.event.inventory.InventoryOpenEvent;
+import org.bukkit.event.server.PluginDisableEvent;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.InventoryView;
+import org.bukkit.plugin.Plugin;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.function.Consumer;
 
 /**
- * Listens to events for {@link Gui}s. Only one of these is ever initialized per plugin.
+ * Listens to events for {@link Gui}s. Only one instance of this class gets constructed.
+ * (One instance per plugin, but plugins are supposed to shade and relocate IF.)
  *
  * @since 0.5.4
  */
 public class GuiListener implements Listener {
 
+    /**
+     * A collection of all {@link Gui} instances that have at least one viewers.
```

Comment:
"viewers" should be "viewer" as far as I'm aware, since the "one" implies a singular noun.